src/view/pane/message.js

```diff
@@ -110,8 +110,20 @@ Candy.View.Pane = (function(self, $) {
         timestamp = Candy.Util.iso8601toDate(timestamp);
       }
 
+      var messagePane = self.Room.getPane(roomJid, ".message-pane");
```

Comment:
yes, I'd go for this option as well. Another solution would be to generate when receiving a unique id based on message body, from jid and date.. but this might generate false duplicates :/